weatherman.py

```diff
@@ -0,0 +1,246 @@
+        minimum_temp_day = 1
+        for month in __weather__["Murree"][year].keys():
+            for day in __weather__["Murree"][year][month].keys():
+                if not minimum_temp:
+                    minimum_temp = __weather__["Murree"][year][month][day]["Min TemperatureC"]
+                if (__weather__["Murree"][year][month][day]["Min TemperatureC"] and __weather__["Murree"][year][month][day][
+                    "Min TemperatureC"] <
+                        minimum_temp):
+                    minimum_temp = __weather__["Murree"][year][month][day]["Min TemperatureC"]
+                    minimum_temp_day = day
+                    minimum_temp_month = month
+        return minimum_temp, minimum_temp_day, minimum_temp_month
+
+    @staticmethod
+    def maximum_humidity(year):
+        max_humidity = 0
+        max_humidity_month = "Jan"
+        max_humidity_day = __weather__["Murree"][year][max_humidity_month][1]
+        for month in __weather__["Murree"][year].keys():
+            for day in __weather__["Murree"][year][month].keys():
+                if (__weather__["Murree"][year][month][day]["Max Humidity"] and __weather__["Murree"][year][month][day][
+                    "Max Humidity"] >
+                        max_humidity):
+                    max_humidity = __weather__["Murree"][year][month][day]["Max Humidity"]
+                    max_humidity_month = month
+                    max_humidity_day = day
+        return max_humidity, max_humidity_month, max_humidity_day
+
+    @staticmethod
+    def average_highest_temperature(year, month):
+        average_highest_temp = 0
+        no_of_months = len(__weather__["Murree"][year][month].keys())
+        for day in __weather__["Murree"][year][month].keys():
+            average_highest_temp += __weather__["Murree"][year][month][day]["Max TemperatureC"]
+        return average_highest_temp//no_of_months
+
+    @staticmethod
+    def average_lowest_temperature(year, month):
+        average_lowest_temp = 0
+        no_of_months = len(__weather__["Murree"][year][month].keys())
+        for day in __weather__["Murree"][year][month].keys():
+            average_lowest_temp += __weather__["Murree"][year][month][day]["Min TemperatureC"]
+        return average_lowest_temp//no_of_months
+
+    @staticmethod
+    def average_mean_humidity(year, month):
+        avg_mean_humidity = 0
+        no_of_months = len(__weather__["Murree"][year][month].keys())
+        for day in __weather__["Murree"][year][month].keys():
+            avg_mean_humidity += __weather__["Murree"][year][month][day]["Mean Humidity"]
+        return avg_mean_humidity//no_of_months
+
+    @staticmethod
+    def maximum_temperature_on_day(year, month, day):
+        return __weather__["Murree"][year][month][day]["Max TemperatureC"]
+
+    @staticmethod
+    def minimum_temperature_on_day(year, month, day):
+        return __weather__["Murree"][year][month][day]["Min TemperatureC"]
```

Comment:
Revisit approach of all computations here, most of the detailed comments are made in one single method with the hope that you will be fixing it in other areas as well.